docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,237 @@
+containers.  For Clear Containers, we have enabled a CNM plugin and necessary
+changes in the runtime to detect virtual functions in a container's 
+network namespace to facilitate making use of SR-IOV for network based devices.
+
+## Install the SR-IOV Docker\* plugin
+
+In order to create a network with VFs associated with it which can be passed to
+Clear Containers, it is necessary to install a SR-IOV Docker plugin. The 
+created network will be based on a physical function (PF) device, and can
+create 'n' containers, where 'n' is the number of VFs associated with the 
+Physical Function (PF).
+
+To install the plugin, follow the [plugin installation instructions](https://github.com/clearcontainers/sriov).
+
+
+## Host setup for SR-IOV
+
+You may need to rebuild your host system's kernel in order to disable 
+`CONFIG_VFIO_NOIOMMU` in the kernel configuration and potentially add a PCI 
+override for your Network Interface Controller (NIC). If not, you can
+move to the next section. The section below describes how to assess if
+changes are needed, and how to make these changes.
+
+### IOMMU Groups and PCIe Access Control Services
+Taking a look at how the Input-Output Memory Management Unit (IOMMU) groups are
+setup on your host system can help provide information on whether or not your 
+NIC is setup appropriately with respect to PCIe Access Control Services (ACS).
+More specifically, if the PCI bridge is within the same IOMMU group as your 
+NIC, then this is an indication that either your device doesn't support ACS or 
+that it doesn't share this capability appropriately by default.
+
+For example, when you run the following command, if all is setup properly, you
+should have the PCI for each ACS enabled NIC port in its own IOMMU group. If
+you don't see any output when running the following, then you likely need to
+update your host's kernel configuration to disable `CONFIG_VFIO_NOIOMMU`.
+
+```
+$ find /sys/kernel/iommu_groups/ -type l
+```
+
+For more details, see http://vfio.blogspot.com/2014/08/iommu-groups-inside-and-out.html
+
+### Updating the Host Kernel
+
+These directions are provided as an example, and are based on Ubuntu 16.04. The
+user will need to do something equivalent for their distribution.
+
+1.  Grab kernel sources
+
+```
+$ sudo apt-get install linux-source-4.10.0
+$ sudo apt-get install linux-headers-4.10.0
+$ cd /usr/src/linux-source-4.10.0/
+$ sudo tar -xvf linux-source-4.10.0.tar.bz2
+$ cd linux-source-4.10.0
+$ sudo apt-get install libssl-dev
+```
+
+2. Check the config and update if necessary
+
+```
+$ sudo cp /boot/config-4.8.0-36-generic .config
+$ # verify resulting .config does not have NOIOMMU set; ie: `CONFIG_VFIO_NOIOMMU` is not set
+$ # verify `CONFIG_VFIO_IOMMU_TYPE1`, `CONFIG_VFIO=m` and `CONFIG_VFIO_PCI=m` are set as well.
+$ sudo make olddefconfig 
+```
+
+You may choose to modify the kernel `Makefile` to add a unique indentifier 
```

Comment:
Can we add a reason they would they choose that? If it is recommended, lets just say: To modify..

Reply:
The reason is mentioned here , so that when the kernel is running, the command `uname -r` indicates a customised kernel is being run.